tests/benchq/compilation/test_graph_sim_mini.py

```diff
@@ -0,0 +1,180 @@
+
+
+def get_target_tableau(circuit):
+    sim = stim.TableauSimulator()
+    for op in circuit.operations:
+        if op.gate.name == "I":
+            pass
+        elif op.gate.name == "CNOT":
+            sim.cx(*op.qubit_indices)
+        elif op.gate.name == "S_Dagger":
+            sim.s_dag(*op.qubit_indices)
+        else:
+            getattr(sim, op.gate.name.lower())(*op.qubit_indices)
+    return get_tableau_from_stim_simulator(sim)
+
+
+def get_stabilizer_tableau_from_vertices(vertices):
+    n_qubits = len(vertices)
+
+    all_xs = np.identity(n_qubits, dtype=bool)
+    all_zs = np.zeros((n_qubits, n_qubits), dtype=bool)
+
+    for vertex_id, vertex in enumerate(vertices):
+        for neighbor in vertex[1]:
+            all_zs[neighbor, vertex_id] = True
+            all_zs[vertex_id, neighbor] = True
+
+    paulis = []
+    for xs, zs in zip(all_xs, all_zs):
+        paulis = paulis + [stim.PauliString.from_numpy(xs=xs, zs=zs)]
+
+    sim = stim.TableauSimulator()
+    tableau = stim.Tableau.from_stabilizers(paulis)
+    sim.set_inverse_tableau(tableau.inverse())
+
+    cliffords = []
+    for vertex in vertices:
+        # perform the vertex operation on the tableau
+        pauli_perm_class = divmod(vertex[0], 4)[0]
+        if pauli_perm_class == 0:
+            cliffords += [[]]
+        if pauli_perm_class == 1:
+            cliffords += [["s"]]
+        if pauli_perm_class == 2:
+            cliffords += [["h"]]
+        if pauli_perm_class == 3:
+            cliffords += [["h", "s", "h"]]
+        if pauli_perm_class == 4:
+            cliffords += [["s", "h"]]
+        if pauli_perm_class == 5:
+            cliffords += [["h", "s"]]
+
+    # perform the vertices operations on the tableau
+    for i in range(n_qubits):
+        for clifford in cliffords[i]:
+            getattr(sim, clifford)(i)
+
+    return get_tableau_from_stim_simulator(sim)
+
+
+def get_tableau_from_stim_simulator(sim):
+    return np.column_stack(sim.current_inverse_tableau().inverse().to_numpy()[2:4])
+
+
+def tableaus_correspond_to_same_state(state_1, state_2):
+    for stab_1 in state_1:
+        for stab_2 in state_2:
+            if not commutes(stab_1, stab_2):
+                return False
+    return True
+
+
+def commutes(stab_1, stab_2):
+    """Returns true if self commutes with other, otherwise false.
+
+    Args:
+        other (SymplecticPauli): SymplecticPauli for commutation
+
+    Returns:
+        bool: true if self and other commute, false otherwise.
+    """
+    n_qubits = len(stab_1) // 2
+    comm1 = bool_dot(stab_1[:n_qubits], stab_2[n_qubits:])
+    comm2 = bool_dot(stab_1[n_qubits:], stab_2[:n_qubits])
+    return not (comm1 ^ comm2)
+
+
+# numpy doesn't use the boolean binary ring when performing dot products
+# https://github.com/numpy/numpy/issues/1456.
+# So we define our own dot product which uses "xor" instead of "or" for addition.
+def bool_dot(x, y):
```

Comment:
Most of the functions here could start with "_"?
Extreme nitpick ;)

Reply:
eh, just bool dot really needs it. Others might be useful to users.